project-koku/koku
# remove `units` query parameter and refactor serializers

## Description

* This change removes `units` as a query parameter. This is a broken and unused feature of the API. 
* This PR also refactors/reorders our serializers:
  * The ReportQuerySerializers were refactored to move all common funcs into the ReportQuerySerializer base class
  * The Tag serializers were moved to provider specific files to match how the report serializers are organized

## Testing

All of these units as query params are invalid and should return 400 Bad Request:
```
{"units": "%2F%2F8028544394501274830.owasp.org"}
{"units": "%253CscrIpt%253Ealert%25281%2529%253B%253C%252FscRipt%253E"}
{"units": "units%3Bget-help"}
{"units": "units%27%7Ctimeout+%2FT+15"}
{"units": "%5D%5D>"}
```

1. make any reporting API request: [`http://localhost:8000/api/cost-management/v1/reports/{INSERT-REPORTING-ENDPOINT}/?filter=&group_by=&order_by=&units={INSERT_UNITS}&offset=0&limit=100&start_date=1970-01-01&end_date=1970-01-01`](http://localhost:8000/api/cost-management/v1/reports/openshift/infrastructures/aws/instance-types/?filter=&group_by=&order_by=&units=%5D%5D%3E&offset=0&limit=100&start_date=1970-01-01&end_date=1970-01-01)
2. See 500 Status with a 500 Internal Server Error:
```
TokenError at /api/cost-management/v1/reports/openshift/infrastructures/aws/instance-types/
('EOF in multi-line statement', (2, 0))
```
1. Checkout Branch
3. Restart Koku
4. Hit endpoint again and see 400 Status Bad Request:
```
{
    "units": [
        "Unsupported parameter or invalid value"
    ]
}
```
